Indicators/EaseOfMovementValue.cs

```diff
@@ -0,0 +1,102 @@
+﻿/*
+ * QUANTCONNECT.COM - Democratizing Finance, Empowering Individuals.
+ * Lean Algorithmic Trading Engine v2.0. Copyright 2014 QuantConnect Corporation.
+ * 
+ * Licensed under the Apache License, Version 2.0 (the "License"); 
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+*/
+using QuantConnect.Data.Market;
+
+namespace QuantConnect.Indicators
+{
+    /// <summary>
+    /// This indicator computes the n-period Ease of Movement Value using the following:
+    /// MID = (high_1 + low_1)/2 - (high_0 + low_0)/2 
+    /// RATIO = (currentVolume/10000) / (high_1 - low_1)
+    /// EMV = MID/ratio
+    /// </summary>
+    public class EaseOfMovementValue : TradeBarIndicator, IIndicatorWarmUpPeriodProvider
+    {
+
+        private readonly int _period;
+        private readonly IndicatorBase<IndicatorDataPoint> _maximum;
+        private readonly IndicatorBase<IndicatorDataPoint> _minimum;
+
+        public decimal PreviousHighPrice { get; private set; }
+        public decimal PreviousLowPrice { get; private set; }
+
+        /// <summary>
+        /// Gets a flag indicating when this indicator is ready and fully initialized
+        /// </summary>
+        public override bool IsReady => _maximum.IsReady && _minimum.IsReady;
+
+        /// <summary>
+        /// Required period, in data points, for the indicator to be ready and fully initialized.
+        /// </summary>
+        public int WarmUpPeriod { get; }
+
+
+        public override void Reset()
+        {
+            PreviousHighPrice = 1;
+            PreviousLowPrice = 1;
+            _maximum.Reset();
+            _minimum.Reset();
+            base.Reset();
+        }
+
+        /// <summary>
+        /// Initializeds a new instance of the EaseOfMovement class using the specufued period
+        /// </summary>
+        /// <param name="period">The period over which to perform to computation</param>
+        public EaseOfMovementValue(int period = 2)
+            : this($"EMV({period})", period)
+        {
+        }
+
+        /// <summary>
+        /// Creates a new EaseOfMovement indicator with the specified period
+        /// </summary>
+        /// <param name="name">The name of this indicator</param>
+        /// <param name="period">The period over which to perform to computation</param>
+        public EaseOfMovementValue(string name, int period)
+            : base(name)
+        {
+            WarmUpPeriod = period;
+            _maximum = new Maximum(period);
+            _minimum = new Minimum(period);
```

Comment:
You could use
```csharp
_previousHighMaximum = new Delay(1).MAX(period);
_previousLowMinimum = new Delay(1).MIN(period);
```
to eliminate `PreviousHighPrice` and `PreviousLowPrice`. I don't think they should be public members.

Reply:
Thanks Alex. I made all the changes you requested. I also removed the maximum indicator as it was redundant.